### PR TITLE
[TextFields] Better custom fonts

### PR DESCRIPTION
**Pull request by willlarche**
Replaces magic KVO code with public API in the same pattern as the recent API for label colors. Do not worry about bloat; this is the last 'thing' that can be moved to public API per label.

### CODE REVIEW

**Comment by willlarche** [comment]
Closes #1920

File: components/TextFields/src/MDCTextInputControllerDefault.m [hunk 1]
```diff
@@ -972,6 +952,44 @@ + (void)setInlinePlaceholderColorDefault:(UIColor *)inlinePlaceholderColorDefaul
                                        : MDCTextInputDefaultInlinePlaceholderTextColorDefault();
 }
 
+- (UIFont *)inlinePlaceholderFont {
+  return _inlinePlaceholderFont ?: [self class].inlinePlaceholderFontDefault;
+}
+
+- (void)setInlinePlaceholderFont:(UIFont *)inlinePlaceholderFont {
+  if (![_inlinePlaceholderFont isEqual:inlinePlaceholderFont]) {
+    _inlinePlaceholderFont = inlinePlaceholderFont;
```
Comment (randallli) [inline components/TextFields/src/MDCTextInputControllerDefault.m:961]: We are letting the iVar go to nil and assume we will always use the getter?

style guide discourages accessors in some places.
http://google.github.io/styleguide/objcguide.html#avoid-accessors-during-init-and-dealloc

Comment (randallli) [reply components/TextFields/src/MDCTextInputControllerDefault.m:961]: I guess we are already not following the guidance is this component. We should consider doing so.

**Review by randallli — APPROVED** [review]

File: components/TextFields/src/MDCTextInputControllerDefault.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerDefault.m:961]: This will be handled in a future PR.

**Pull request merged**